packages/synapse-react-client/src/components/AccessRequirementList/ManagedACTAccessRequirementRequestFlow/ManagedACTAccessRequirementItemView.tsx

```diff
@@ -0,0 +1,305 @@
+}
+
+const APPROVED_TEXT = 'Your data access request has been approved.'
+const EXEMPT_TEXT =
+  'This data access requirement is fulfilled by your team membership and permissions.'
+
+/**
+ * Translate the ManagedACTAccessRequirementStatus and exemption status into a the
+ * RequirementItemStatus displayed by the UI
+ * @param status
+ * @param isExempt
+ */
+function getRequirementItemUiState(
+  status: ManagedACTAccessRequirementStatus,
+  isExempt: boolean,
+): RequirementItemStatus {
+  if (isExempt) {
+    return RequirementItemStatus.COMPLETE
+  }
+  switch (status.currentSubmissionStatus?.state) {
+    case SubmissionState.APPROVED:
+      return status.isApproved
+        ? RequirementItemStatus.COMPLETE
+        : RequirementItemStatus.LOCKED
+    case SubmissionState.SUBMITTED:
+      return RequirementItemStatus.PENDING
+    case SubmissionState.REJECTED:
+    case SubmissionState.CANCELLED:
+    default:
+      return RequirementItemStatus.LOCKED
+  }
+}
+
+/**
+ * Gets the action buttons to display for a ManagedACTAccessRequirement based on the current state of the requirement.
+ * @param accessRequirementStatus - the user's status for the access requirement
+ * @param isExempt - whether the user is exempt from submitting a request for this access requirement
+ * @param cancelRequestIsPending - whether a request to cancel the data access request is currently pending
+ * @param callbacks - callbacks to invoke when the user clicks the action buttons
+ */
+function getActionButtonProps(
+  accessRequirementStatus: ManagedACTAccessRequirementStatus,
+  isExempt: boolean,
+  cancelRequestIsPending: boolean,
+  callbacks: {
+    onRequestAccess: () => void
+    onRejectTerms: () => void
+    onCancelDataAccessRequest: () => void
+  },
+): ButtonProps[] {
+  const { onRequestAccess, onRejectTerms, onCancelDataAccessRequest } =
+    callbacks
+
+  const actions: ButtonProps[] = []
+
+  if (accessRequirementStatus) {
+    if (accessRequirementStatus.currentSubmissionStatus == null) {
+      actions.push({
+        variant: 'outlined',
+        onClick: onRequestAccess,
+        children: 'Request access',
+      })
+      if (!isExempt) {
+        actions.push({
+          variant: 'text',
+          onClick: onRejectTerms,
+          children: 'I do not accept',
+        })
+      }
+    } else if (
+      [
+        SubmissionState.APPROVED,
+        SubmissionState.REJECTED,
+        SubmissionState.CANCELLED,
+      ].includes(accessRequirementStatus.currentSubmissionStatus.state)
+    ) {
+      actions.push({
+        variant: 'outlined',
+        onClick: onRequestAccess,
+        children: 'Update Request',
+      })
+    } else if (
+      accessRequirementStatus.currentSubmissionStatus.state ===
+      SubmissionState.SUBMITTED
+    ) {
+      actions.push({
+        variant: 'outlined',
+        onClick: onCancelDataAccessRequest,
+        children: 'Cancel Request',
+        loading: cancelRequestIsPending,
```

Comment:
New: handle state where user initiated cancel but the server has not yet responded (using new MUI Button `loading` prop)

![image](https://github.com/user-attachments/assets/18feb2f3-a552-4116-a042-0c47b6d3c81d)